portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,6 +35,9 @@
 /** Servlet that encapsulates comments. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
+  String userFilter;
+  int commentNumber;
```

Comment:
These variables don't need to be class members.

Instead try to define them in the smaller scope where they are needed (in the method).

How would that work if you have multiple different users calling doGet?

How does multi-threading for Java Servlets work? Are multiple threads allowed to call doGet concurrently? If yes, then this is unsafe.